c10/BUCK.oss

```diff
@@ -8,6 +8,7 @@ cxx_library(
             "test/**/*.cpp",
             "benchmark/**/*.cpp",
             "cuda/**/*.cpp",
+            "xpu/**/*.cpp",
```

Comment:
@malfet Sorry, I skip buck build for xpu here. Please help review.

Reply:
@malfet , the result shows windows and ROCM build failures. But I think this PR should not trigger the two failures. I'm not sure if we missed something. May I know your insights on whether we need to triage the failures beyond BUCK build?